### PR TITLE
Kernel: Fixed fdc read/write problem

### PR DESCRIPTION
**Pull request by Quaker762**
It seems ~the compiler might have been doing something strange to
the calls to `send_byte()`~ that operator precedence eludes me. Whenever the value of `head` was 1,
`is_slave()` would also be 1 due to abuse of the ternary operator
This resulted in the controller failing the read/write.

I've also added a seek to the top of the read/write code, which seems
to have fixed an issue with Linux not detecting the disk images after
they have been unmounted from Serenity. This isn't specified in the
datasheet, but a few other drivers have it so we should too :^)

### CODE REVIEW

**Comment by awesomekling** [comment]
>blaming the compiler for operator precedence

File: Kernel/Devices/FloppyDiskDevice.cpp [hunk 1]
```diff
@@ -124,7 +124,8 @@ bool FloppyDiskDevice::read_sectors_with_dma(u16 lba, u16 count, u8* outbuf)
 
     motor_enable(is_slave() ? 1 : 0); // Should I bother casting this?!
     write_ccr(0);
-    recalibrate(); // Recalibrate the drive
+    recalibrate();
+    ASSERT(seek(lba));
```
Comment (awesomekling) [inline Kernel/Devices/FloppyDiskDevice.cpp]: Let's not do effectful things in `ASSERT` macros, since they will be compiled out in release builds.

File: Kernel/Devices/FloppyDiskDevice.cpp [hunk 2]
```diff
@@ -193,7 +197,7 @@ bool FloppyDiskDevice::read_sectors_with_dma(u16 lba, u16 count, u8* outbuf)
         static_cast<void>(pcn);
 
         memcpy(outbuf, m_dma_buffer_page->paddr().as_ptr(), 512 * count);
-
+
```
Comment (awesomekling) [inline Kernel/Devices/FloppyDiskDevice.cpp]: Unnecessary whitespace change.

File: Kernel/Devices/FloppyDiskDevice.cpp [hunk 3]
```diff
@@ -213,6 +217,7 @@ bool FloppyDiskDevice::write_sectors_with_dma(u16 lba, u16 count, const u8* inbu
     motor_enable(is_slave() ? 1 : 0); // Should I bother casting this?!
     write_ccr(0);
     recalibrate(); // Recalibrate the drive
+    ASSERT(seek(lba));
```
Comment (awesomekling) [inline Kernel/Devices/FloppyDiskDevice.cpp]: Same comment about effectful assertions.

File: Kernel/Devices/FloppyDiskDevice.cpp [hunk 4]
```diff
@@ -230,6 +235,7 @@ bool FloppyDiskDevice::write_sectors_with_dma(u16 lba, u16 count, const u8* inbu
     u16 cylinder = lba2cylinder(lba);
     u16 head = lba2head(lba);
     u16 sector = lba2sector(lba);
+    u8 drivesel = (is_slave() == true) ? 1 : 0;
```
Comment (awesomekling) [inline Kernel/Devices/FloppyDiskDevice.cpp]: This seems excessively verbose.
```suggestion
    u8 drivesel = is_slave();
```

File: Kernel/Devices/FloppyDiskDevice.cpp [hunk 5]
```diff
@@ -238,7 +244,7 @@ bool FloppyDiskDevice::write_sectors_with_dma(u16 lba, u16 count, const u8* inbu
     for (int i = 0; i < 3; i++) {
         // Now actually send the command to the drive. This is a big one!
         send_byte(FLOPPY_MFM | FLOPPY_MT | static_cast<u8>(FloppyCommand::WriteData));
-        send_byte(head << 2 | is_slave() ? 1 : 0);
+        send_byte(head << 2 | drivesel);
```
Comment (awesomekling) [inline Kernel/Devices/FloppyDiskDevice.cpp]: This could also be written this way:
```suggestion
        send_byte(head << 2 | is_slave());
```

**Comment by Quaker762** [comment]
> 
> 
> > blaming the compiler for operator precedence

So `head` was being sent to `send_byte()` as 1 instead of (1 << 2)? I assume the call to `is_slave()` was causing that right?

File: Kernel/Devices/FloppyDiskDevice.cpp (hunk 1 above)
Comment (Quaker762) [reply Kernel/Devices/FloppyDiskDevice.cpp]: Detect it returned `false` and print an error?

Comment (awesomekling) [reply Kernel/Devices/FloppyDiskDevice.cpp]: That'll work :)

**Comment by awesomekling** [comment]
Please also fix the commit message :)

**Comment by awesomekling** [comment]
Cool :)
